src/data/CHDatasource.ts

```diff
@@ -159,15 +191,59 @@ export class Datasource extends DataSourceWithBackend<CHQuery, CHConfig> {
     return [];
   }
 
-  async fetchTags(): Promise<DataFrame> {
-    // @todo https://github.com/grafana/grafana/issues/13109
-    const rawSql = this.templateSrv.replace('$clickhouse_adhoc_query');
-    if (rawSql === '$clickhouse_adhoc_query') {
-      return new ArrayDataFrame([]);
+  private async fetchTags(): Promise<Tags> {
+    const tagSource = this.getTagSource();
+    this.skipAdHocFilter = true;
+
+    if (tagSource.source === undefined) {
+      this.adHocFilter.setTargetTable('default');
+      const rawSql = 'SELECT name, type, table FROM system.columns';
```

Comment:
Do we want to limit these results too?

Reply:
We may have to add some capability to filter/limit.  They can always override this for now with the helper variable.